packages/@aws-cdk/aws-sagemaker/lib/scalable-instance-count.ts

```diff
@@ -0,0 +1,66 @@
+import * as appscaling from '@aws-cdk/aws-applicationautoscaling';
+import { Construct } from 'constructs';
+
+/**
+ * The properties of a scalable attribute representing task count.
+ */
+export interface ScalableInstanceCountProps extends appscaling.BaseScalableAttributeProps {
+}
+
+/**
+ * A scalable sagemaker endpoint attribute
+ */
+export class ScalableInstanceCount extends appscaling.BaseScalableAttribute {
+  /**
+   * Constructs a new instance of the ScalableInstanceCount class.
+   */
+  constructor(scope: Construct, id: string, props: ScalableInstanceCountProps) {
+    super(scope, id, props);
+  }
+
+  /**
+   * Scales in or out to achieve a target requests per second per instance.
+   */
+  public scaleOnInvocations(id: string, props: InvocationsScalingProps) {
+    const predefinedMetric = appscaling.PredefinedMetric.SAGEMAKER_VARIANT_INVOCATIONS_PER_INSTANCE;
+
+    super.doScaleToTrackMetric(id, {
+      policyName: props.policyName,
+      disableScaleIn: props.disableScaleIn,
+      scaleInCooldown: props.scaleInCooldown,
+      scaleOutCooldown: props.scaleOutCooldown,
+      targetValue: this.calculateScalingTarget(props),
+      predefinedMetric,
+    });
+  }
+
+  /**
+   * Calculate target value based on a ScalableProductionVariant
+   *
+   * Documentation for the equation is here: https://docs.aws.amazon.com/sagemaker/latest/dg/endpoint-scaling-loadtest.html
+   * @param scalableProductionVariant ScalableProductionVariant instance
+   */
+  private calculateScalingTarget( props: InvocationsScalingProps): number {
+    const safetyFactor = props.safetyFactor || 0.5;
```

Comment:
are there any sane guardrails we should be putting on the safetyFactor? Is there a use case for someone to make `safetyFactor > 1`? Or 0? A few of these edge cases would result in `calculateScalingTarget` returning some problematic info, whether thats ridiculously high or low.